packages/amplify-migration-template-gen/src/resolvers/cfn-output-resolver.ts

```diff
@@ -15,10 +15,13 @@ class CfnOutputResolver {
   public resolve(logicalResourceIds: string[], stackOutputs: Output[]): CFNTemplate {
     const resources = this.template?.Resources;
     assert(resources);
-    let stackTemplateString = JSON.stringify(this.template);
+    const clonedStackTemplate = JSON.parse(JSON.stringify(this.template)) as CFNTemplate;
     const stackTemplateOutputs = this.template?.Outputs;
+    const stackTemplateResources = this.template?.Resources;
+    assert(stackTemplateResources);
     assert(stackOutputs);
     assert(stackTemplateOutputs);
+    let stackTemplateResourcesString = JSON.stringify(stackTemplateResources);
```

Comment:
the main change here is to only replace Refs within Resources rather than the entire template.